cpp/memilio/epidemiology/populations.h

```diff
@@ -253,8 +252,9 @@ class Populations : public CustomIndexArray<UncertainValue, Categories...>
     void check_constraints() const
     {
         for (int i = 0; i < this->array().size(); i++) {
-            if (this->array()[i] < 0) {
-                log_error("Constraint check: Compartment size {:d} is {:.4f} and smaller {:d}", i, this->array()[i], 0);
+            FP value = this->array()[i];
+            if (value < 0.) {
+                log_error("Constraint check: Compartment size {:d} is {:.4f} and smaller {:d}", i, value, 0);
```

Comment:
```suggestion
                log_error("Constraint check: Compartment size {} is {} and smaller {}", i, value, 0);
```

Reply:
It was correct, wasn't it?

Reply:
Kind of. I had problems with the :d format before, it seems neither me nor spdlog really know what it is supposed to do. And I don't know of any drawbacks from just letting spdlog figure out the correct type and format.